pkg/util/ec2/ec2.go

```diff
@@ -48,6 +48,29 @@ func getMetadataItem(endpoint string) (string, error) {
 	return string(all), nil
 }
 
+// GetClusterName returns the name of the cluster containing the current EC2 instance
+func GetClusterName() (string, error) {
+	tags, err := GetTags()
```

Comment:
What's the rationale behind calling `GetTags` here?

Reply:
as in EKS we retrieve `clustername` from tags I found it maybe an overkill to refactor the `ec2_tags.go` module, we can add a `GetClusterNameTag` function and refactor the `GetTags` function and separate the session creation logic in a `CreateEC2Session` function to reduce code duplication then call it in `GetTags` and `GetClusterNameTag`, I find it simpler to call `GetTags` the way I did it but if you think the opposite I am happy to change it ;)

Reply:
LGTM, as long as:
  - we bubble up retrieval errors correctly to the agent logs
  - the tag_list -> cluster_name logic is covered by unit tests

Reply:
okay so i'll add  tag_list -> cluster_name logic unit test